core/include/detray/grids/grid2.hpp

```diff
@@ -338,13 +344,21 @@ class grid2 {
  * @tparam tparam axis_p1_type the type of the second axis
  * @tparam serialzier_type  type of the serializer to the storage represenations
  **/
-template <typename grid_t>
+template <template <template <typename...> class, template <typename...> class,
+                    template <typename, unsigned int> class, typename, auto...>
+          class populator_type,
+          typename axis_p0_type, typename axis_p1_type,
+          typename serializer_type,
+          template <typename...> class vector_type = dvector,
+          template <typename...> class jagged_vector_type = djagged_vector,
+          template <typename, unsigned int> class array_type = darray,
+          template <typename...> class tuple_type = dtuple,
+          typename value_type = dindex, auto... populator_options>
 struct grid2_buffer {
 
-    using populator_t = typename grid_t::populator_t;
-    using serializer_t = typename grid_t::serializer_t;
-    using axis_p0_t = typename grid_t::axis_p0_t;
-    using axis_p1_t = typename grid_t::axis_p1_t;
+    using populator_t =
+        populator_type<vector_type, jagged_vector_type, array_type, value_type,
+                       populator_options...>;
```

Comment:
Strange... Did the previous formalism not work with the new setup of `detray::grid2`? :thinking:

In algebra-plugins I "inherit" some complicated types in the same way as this class used to do. For instance:

https://github.com/acts-project/algebra-plugins/blob/main/math/cmath/include/algebra/math/impl/cmath_cartesian2.hpp#L23-L29

On first sight I would've thought hat it would still be possible to "inherit" the `populator_t` type from an "upstream" `grid_t` type, as the code used to do. Reducing the number of places where this complicated signature would have to be written out.

Can you give this a try? To see if it could still be possible to template this class on a single `grid_t` type?